src/core/sock.c

```diff
@@ -292,6 +298,12 @@ static int nn_sock_setopt_inner (struct nn_sock *self, int level,
         return 0;
     }
 
+    if (level == NN_SOL_SOCKET && option == NN_SEC_ATTR) {
```

Comment:
NN_SOL_SOCKET feels like the wrong socket option for this.  Its a transport specific option, so you should define one.  Its kind of unfortunate that we need this at all, as Windows is quite different from UNIX in this regard.

For UNIX we're punting, requiring users to change the permissions of any socket after creation.  Is this possible with Windows?